core/src/main/java/io/questdb/griffin/engine/ops/CreateMatViewOperationImpl.java

```diff
@@ -473,39 +503,54 @@ private static void copyBaseTableColumnNames(
                     if (dotIndex > -1) {
                         if (Chars.equalsIgnoreCase(model.getName(), node.token, 0, dotIndex)) {
                             if (!Chars.equalsIgnoreCase(model.getTableName(), baseTableName)) {
-                                throw SqlException.$(node.position, "only base table columns can be used as keys: ").put(node.token);
+                                throw SqlException.position(node.position + selectTextPosition)
+                                        .put("only base table columns can be used as materialized view keys")
+                                        .put(" [invalid key=").put(node.token)
+                                        .put(']');
                             }
-                            target.add(Chars.toString(node.token, dotIndex + 1, node.token.length()));
+                            baseKeyColumnNames.add(Chars.toString(node.token, dotIndex + 1, node.token.length()));
+                            baseKeyColumnNamePositions.add(node.position);
                             return;
                         }
                     } else {
                         if (!Chars.equalsIgnoreCase(model.getTableName(), baseTableName)) {
-                            throw SqlException.$(node.position, "only base table columns can be used as keys: ").put(node.token);
+                            throw SqlException.position(node.position + selectTextPosition)
+                                    .put("only base table columns can be used as materialized view keys")
+                                    .put(" [invalid key=").put(node.token)
+                                    .put(']');
                         }
-                        target.add(node.token);
+                        baseKeyColumnNames.add(Chars.toString(node.token));
+                        baseKeyColumnNamePositions.add(node.position);
                         return;
                     }
                 } else {
                     // Check nested model.
                     final QueryColumn column = model.getAliasToColumnMap().get(node.token);
-                    copyBaseTableColumnNames(column != null ? column.getAst() : node, model.getNestedModel(), baseTableName, target);
+                    copyBaseTableColumnNames(column != null ? column.getAst() : node,
```

Comment:
nit: staircase formatting

Reply:
fixed in https://github.com/questdb/questdb/pull/5491/commits/6065f7b26ea6fb7d33cd48b37eed762204729ad0